gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/web/internal/configureWebApplication.kt

```diff
@@ -86,6 +84,98 @@ internal fun configureWebApplication(
         outputDir.set(unpackedRuntimeDir)
     }
 
+    val webProductionDist = project.layout.buildDirectory.dir("dist/web/productionExecutable")
+    val jsAppRenamed = "__jsApp.js"
+    val wasmAppRenamed = "__wasmApp.js"
+
+    val copyJsDistSpec = project.copySpec { spec ->
+        val jsTarget = targets.firstOrNull { it.name == "js" }?.outputModuleName
+        spec.apply {
+            from( project.tasks.named("jsBrowserDistribution")) {
+                rename { name ->
+                    val moduleName = jsTarget?.get()
+                    when (name) {
+                        "${moduleName}.js" -> jsAppRenamed
+                        "${moduleName}.js.map" -> "${jsAppRenamed}.map"
+                        else -> name
+                    }
+                }
+            }
+        }
+    }
+
+    val copyWasmDistSpec = project.copySpec { spec ->
+        val wasmTarget = targets.firstOrNull { it.name == "wasmJs" }?.outputModuleName
+        spec.apply {
+            from( project.tasks.named("wasmJsBrowserDistribution")) {
+                rename { name ->
+                    val moduleName = wasmTarget?.get()
+                    when (name) {
+                        "${moduleName}.js" -> wasmAppRenamed
+                        "${moduleName}.js.map" -> "${wasmAppRenamed}.map"
+                        else -> name
+                    }
+                }
+            }
+        }
+    }
+
+
+    val composeWebCompatibilityDistTask = project.registerTask<Copy>("composeWebCompatibilityDist") {
```

Comment:
this task is registered unconditionally. 

This task only makes sense if there are both k/js and k/wasm targets.

Reply:
I see there is 

```
onlyIf {
            jsTarget != null && wasmTarget != null
        }
```

But for a user of this gradle task it's not clear why it's available but doesn't run.

Reply:
@eymar yep, that's the idea - this way it's more obvious that this task exists but not running